crates/aggregator/src/lib.rs

```diff
@@ -0,0 +1,312 @@
+                }
+            }
+        });
+        let socket: SocketAddr = aggregator.lock().await.port_address.parse().map_err(|e| {
+            AggregatorError::IOError(std::io::Error::new(std::io::ErrorKind::InvalidInput, e))
+        })?;
+        let server = ServerBuilder::new(io)
+            .cors(DomainsValidation::AllowOnly(vec![
+                AccessControlAllowOrigin::Any,
+            ]))
+            .start_http(&socket)?;
+
+        // TODO: move to an async library
+        tokio::task::spawn_blocking(move || server.wait());
+
+        info!("Server running at {socket}");
+
+        Ok(())
+    }
+
+    /// Processes the tasks
+    ///
+    /// # Arguments
+    ///
+    /// * `ws_rpc_url` - The websocket RPC URL
+    /// * `aggregator` - The aggregator
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), AggregatorError>` - The result of the operation
+    async fn process_tasks(
+        ws_rpc_url: String,
+        aggregator: Arc<Mutex<Self>>,
+    ) -> Result<(), AggregatorError> {
+        let ws = WsConnect::new(ws_rpc_url.clone());
+        let provider = ProviderBuilder::new().on_ws(ws).await?;
+
+        let filter = Filter::new().event_signature(TP::NewTaskEvent::SIGNATURE_HASH);
+        let sub = provider.subscribe_logs(&filter).await?;
+        let mut stream = sub.into_stream();
+
+        while let Some(log) = stream.next().await {
+            let event: TP::NewTaskEvent = log.log_decode()?.inner.data;
+
+            let info = aggregator.lock().await.tp.process_new_task(event).await?;
+
+            aggregator
+                .lock()
+                .await
+                .bls_aggregation_service
+                .initialize_new_task(
+                    info.task_index,
+                    info.task_created_block,
+                    info.quorum_nums,
+                    info.quorum_threshold_percentages,
+                    info.time_to_expiry,
+                )
+                .await?;
+        }
+
+        Ok(())
+    }
+
+    /// Processes the signed task response
+    ///
+    /// # Arguments
+    ///
+    /// * [`SignedTaskResponse`] - The signed task response
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), AggregatorError>` - The result of the operation
+    async fn process_signed_task_response(
+        &mut self,
+        signed_task_response: SignedTaskResponse<TP::TaskResponse>,
+    ) -> Result<(), AggregatorError> {
+        let SignedTaskResponse {
+            task_response,
+            signature,
+            operator_id,
+        } = signed_task_response;
+        let task_index = task_response.task_index();
+
+        let task_response_digest = self.tp.process_task_response(task_response).await?;
+
+        self.bls_aggregation_service
+            .process_new_signature(task_index, task_response_digest, signature, operator_id)
+            .await?;
+        info!("processed signature for index {:?}", task_index);
+        let quorum_reached = {
```

Comment:
this assumed that 2 signatures are needed to process .
there's a better way in dev branch here
https://github.com/Layr-Labs/incredible-squaring-avs-rs/blob/b953a8127b604d91175d71e7c2b47ad4023f7e77/crates/aggregator/src/lib.rs#L320-L395

it handles all cases using total stake and quorum threshold percentage. Its hardcoded rn there with 4800 (40% of 12000(total stake))

Reply:
Yes. However, I was thinking of just waiting for a signature aggregate to be returned by the BLS aggregation service.